### DISCUSSION TITLE
Abaqus and CalculiX umat for CalculiX

### DISCUSSION THREAD

**Discussion by alvlk** [thelfer/tfel#61] [Q&A] [answered]
Hello!

I've tried to use some umat from TFel-Mfront. All ok, but there is an few error, may be many people didn't catch it early, I think. 

I did try to enable Abaqus and CalculiX in the same time, for example

cmake $TFELHOME/$tfelver -Denable-fortran=ON -Denable-abaqus=ON -Denable-calculix=ON -Denable-python=OFF  -DCMAKE_INSTALL_PREFIX=$TFELHOME 

Then it was compiled, installed and builded with some umat for calculix for example with 

./bin/mfront --obuild --interface=abaqus ./share/doc/mfront/tests/behaviours/SaintVenantKirchhoffElasticity.mfront

It works fine in CalculiX. Then I did try to compile this umat for calculix

./bin/mfront --obuild --interface=calculix ./share/doc/mfront/tests/behaviours/SaintVenantKirchhoffElasticity.mfront

And the error was: 

./bin/mfront --obuild --interface=calculix ./share/doc/mfront/tests/behaviours/SaintVenantKirchhoffElasticity.mfront
Treating target : all
In file included from ../include/MFront/Abaqus/AbaqusInterface.hxx:36:0,
                 from abaqusSaintVenantKirchhoffElasticity.cxx:14:
../include/MFront/Abaqus/AbaqusBehaviourHandler.hxx: In instantiation of ‘abaqus::AbaqusBehaviourHandler<H, Behaviour>::Integrator::Integrator(const abaqus::AbaqusData&) [with tfel::material::ModellingHypothesis::Hypothesis H = (tfel::material::ModellingHypothesis::Hypothesis)2; Behaviour = tfel::material::SaintVenantKirchhoffElasticity]’:
../include/MFront/Abaqus/AbaqusInterface.hxx:114:67:   required from ‘static void abaqus::AbaqusInterface<H, Behaviour>::CallBehaviour::exe(const abaqus::AbaqusData&) [with tfel::material::ModellingHypothesis::Hypothesis H = (tfel::material::ModellingHypothesis::Hypothesis)2; Behaviour = tfel::material::SaintVenantKirchhoffElasticity]’
../include/MFront/Abaqus/AbaqusInterface.hxx:70:21:   required from ‘static int abaqus::AbaqusInterface<H, Behaviour>::exe(const abaqus::AbaqusData&) [with tfel::material::ModellingHypothesis::Hypothesis H = (tfel::material::ModellingHypothesis::Hypothesis)2; Behaviour = tfel::material::SaintVenantKirchhoffElasticity]’
abaqusSaintVenantKirchhoffElasticity.cxx:180:129:   required from here
../include/MFront/Abaqus/AbaqusBehaviourHandler.hxx:227:23: error: no matching function for call to ‘tfel::material::SaintVenantKirchhoffElasticity<(tfel::material::ModellingHypothesis::Hypothesis)2, double, false>::SaintVenantKirchhoffElasticity(const AbaqusReal*, const AbaqusReal* const&, const AbaqusReal* const&, const AbaqusReal*, abaqus::AbaqusReal* const&, const AbaqusReal* const&, const AbaqusReal* const&, const AbaqusReal* const&)’
             dt(d.DTIME) {
                       ^
compilation terminated due to -Wfatal-errors.
Makefile.mfront:39: recipe for target 'abaqusSaintVenantKirchhoffElasticity.o' failed
make: *** [abaqusSaintVenantKirchhoffElasticity.o] Error 1
callMake: can't build target 'all'
libraries building went wrong
Command was: make -C src -f Makefile.mfront all -s

If anybody changes the first stage with umat-lib to calculix support, the error was after trying to build umat with abaqus support. I can build TFel only with -Denable-abaqus=ON or -Denable-calculix=ON, not both in the same code.

Thanks for your work.
Best regards,
AlexKaz

**Comment by thelfer** [comment, marked as answer]
Hi @alvlk,

Thanks for your interest to `TFEL/MFront`. Hope you'll enjoy using it !

You encountered an unfortunate and well known caveat: you can't compile the same behaviour with two different interfaces in the same directory (or you shall remove the `include` and `src` directory).

Thomas
